packit/upstream.py

```diff
@@ -249,47 +248,34 @@ def create_patches(
         )
 
         patch_list = []
-        # we need to prefix patches so we can order them properly
-        # this is needed for merge commits - git orders every instance of `format-patch` call
-        commits_count = len(commits)
-        # how many digits does commits_count have?
-        order = floor(log10(commits_count)) + 1
-        patch_counter = 0  # prefix for patches so they are nicely ordered
-        # first value is upstream ref, i.e parent for the first commit we want to create patch from
-        for i, commit in enumerate(commits[1:]):
-            parent_commit = commits[i]
-            git_diff_cmd = [
-                "git",
-                "format-patch",
-                "--output-directory",
-                f"{destination}",
-                f"{parent_commit.hexsha}..{commit.hexsha}",
-                "--",
-                ".",
-            ] + [f":(exclude){file_to_ignore}" for file_to_ignore in files_to_ignore]
-            git_format_patch_out = run_command(
-                cmd=git_diff_cmd,
-                cwd=self.local_project.working_dir,
-                output=True,
-                decode=True,
-            )
 
-            # there can be multiple patches in a merge request
-            # so `git format-patch` can contain multiple files in the output
-            if git_format_patch_out:
-                for patch_file_str in git_format_patch_out.split("\n"):
-                    if not patch_file_str:
-                        continue
-                    prefix = format(patch_counter, f"0{order}")
-                    patch_file = Path(patch_file_str)
-                    patch_name = f"{prefix}-{patch_file.name}"
-                    patch_target_path = Path(destination).joinpath(patch_name)
-                    patch_file.replace(patch_target_path)
-                    msg = f"{commit.summary}\nAuthor: {commit.author.name} <{commit.author.email}>"
-                    patch_list.append((patch_target_path, msg))
-                    patch_counter += 1
-            else:
-                logger.info(f"No patch for commit: {commit.summary} ({commit.hexsha})")
+        git_diff_cmd = [
+            "git",
+            "format-patch",
+            "--output-directory",
+            f"{destination}",
+            upstream,
```

Comment:
Shouldn't this be `f"{upstream}..HEAD"` to be a revision range?

Reply:
they are identical https://stackoverflow.com/a/18680080/909579